### PR TITLE
support for princeton mode

### PR DESCRIPTION
**Pull request by codelec**
emulator not printing the stats after bmark test being the only feature that is not working

### CODE REVIEW

**Comment by ccelio** [comment]
Poke me on this in ~2 weeks. Sorry I'm a bit swamped right now.

File: src/rv32_3stage/arbiter.scala [hunk 1]
```diff
@@ -25,56 +24,58 @@ class SodorMemArbiter(implicit val conf: SodorConfiguration) extends Module
...
-   // default
-   req_fire_dmem := Bool(false) 
-   req_fire_imem := Bool(false) 
-
-   when (io.dmem.req.valid)
+   val i1reg = Reg(Bits(width=conf.xprlen))
+   val d1reg = Reg(Bits(width=conf.xprlen))
+   val nextdmem = Reg(init=Bool(true))
+   io.dmem.req.ready := Bool(true)
+   val d_or_i = new Bool()
```
Comment (ccelio) [inline src/rv32_3stage/arbiter.scala]: What does the `d_or_i` variable do? What does it mean when it is true (`dmem & nextdmem`?)? I'm finding the name confusing. Is something like `d_wins` or `d_fire` more clear?

**Comment by codelec** [comment]
changes made

**Comment by ccelio** [comment]
Thanks for the PR. I really appreciate it!

**Pull request merged**